src/backend/distributed/commands/utility_hook.c

```diff
@@ -344,26 +343,19 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 			}
 			ResetConstraintDropped();
 
+			/*
+			 * We're only interested in top-level CREATE TABLE commands
+			 * to create a tenant schema table or a Citus managed table.
+			 */
 			if (context == PROCESS_UTILITY_TOPLEVEL &&
-				ShouldAddNewTableToMetadata(parsetree))
+				(IsA(parsetree, CreateStmt) ||
+				 IsA(parsetree, CreateForeignTableStmt)))
 			{
-				/*
-				 * Here we need to increment command counter so that next command
-				 * can see the new table.
-				 */
-				CommandCounterIncrement();
-				CreateStmt *createTableStmt = (CreateStmt *) parsetree;
-				Oid relationId = RangeVarGetRelid(createTableStmt->relation,
-												  NoLock, false);
+				CreateStmt *baseCreateTableStmt =
+					IsA(parsetree, CreateStmt) ? (CreateStmt *) parsetree :
+					(CreateStmt *) &(((CreateForeignTableStmt *) parsetree)->base);
```

Comment:
maybe add a comment on why this is not just casting to CreateStmt

(my guess is that this protects us from future postgres changes?)

Reply:
bump, why is this not just `CreateStmt *baseCreateTableStmt = (CreateStmt *) parsetree;` ? could use a comment

Reply:
> (my guess is that this protects us from future postgres changes?)

true, adding a comment for that